cmd/buildtools/addon.go

```diff
@@ -7,24 +7,48 @@ import (
 )
 
 type addonComponent struct {
-	getWolfiPackageName              func(k0sVersion *semver.Version, upstreamVersion *semver.Version) string
-	getWolfiPackageVersionComparison func(k0sVersion *semver.Version, upstreamVersion *semver.Version) string
+	getWolfiPackageName              func(opts commonOptions) string
+	getWolfiPackageVersionComparison func(opts commonOptions) string
 	upstreamVersionInputOverride     string
 }
 
-func (c *addonComponent) getPackageNameAndVersion(wolfiAPKIndex []byte, k0sVersion *semver.Version, upstreamVersion string) (string, string, error) {
+type commonOptions struct {
```

Comment:
you should make this more specific since this package is not. maybe `addonComponentOptions`

Reply:
updated